### PR TITLE
feat(solid): add support for devtools

### PR DESCRIPTION
**Pull request by florian-lefebvre**
## Changes

- Adds a flag to enable https://github.com/thetarnav/solid-devtools in dev mode

## Testing

Manually in the solid example

## Docs

https://github.com/withastro/docs/pull/8189

### CODE REVIEW

File: .changeset/olive-bags-drive.md [hunk 1]
```diff
@@ -0,0 +1,18 @@
+---
+"@astrojs/solid-js": minor
+---
+
+Adds a `devtools` option
+
+Enabling `devtools` will enable the [official Solid Devtools](https://github.com/thetarnav/solid-devtools) in development:
```
Comment (sarah11918) [inline .changeset/olive-bags-drive.md:7]:
```suggestion
You can enable the [official Solid Devtools](https://github.com/thetarnav/solid-devtools) while working in development mode by setting `devtools:true` in your `solid()` integration config:
```

**Review by sarah11918 — COMMENTED** [review]
Just getting ahead of the curve and matching the language of the Vue devtools PR here!

File: .changeset/olive-bags-drive.md (hunk 1 above)
Comment (florian-lefebvre) [reply .changeset/olive-bags-drive.md:7]: Thanks! We'll do the since thing don't worry

**Comment by florian-lefebvre** [comment]
Failing test (sourcemap) looks unrelated, the fixture is not using solid at all

**Review by sarah11918 — APPROVED** [review]
Approving changeset for docs!

**Comment by ematipico** [comment]
@florian-lefebvre I removed the milestone. We use milestones only for core minors, and as a tool to manage PRs

File: .changeset/olive-bags-drive.md [hunk 2]
```diff
@@ -0,0 +1,18 @@
+---
+"@astrojs/solid-js": minor
+---
+
+Adds a `devtools` option
+
+You can enable the [official Solid Devtools](https://github.com/thetarnav/solid-devtools) while working in development mode by setting `devtools:true` in your `solid()` integration config:
```
Comment (bluwy) [inline .changeset/olive-bags-drive.md:7]: I think we should also document that you need to install `solid-devtools` too (and in the docs)

**Review by bluwy — APPROVED** [review]

File: .changeset/olive-bags-drive.md (hunk 2 above)
Comment (florian-lefebvre) [reply .changeset/olive-bags-drive.md:7]: You're right, I'm adding it!

**Review by matthewp — APPROVED** [review]

**Comment by shishkin** [comment]
This has been a breaking change on vite@5. solid-devtools depends on outdated solid-start, which requires vite@4 as peer dependency.

**Comment by florian-lefebvre** [comment]
@shishkin Can you open a new issue with a minimal repro and ping me? I'll have a look, I guess it will require an upstream fix

**Comment by shishkin** [comment]
@florian-lefebvre please see https://github.com/withastro/astro/issues/11217

**Pull request merged**